Library/Homebrew/utils/analytics.sh

```diff
@@ -1,30 +1,42 @@
 setup-analytics() {
-  # User UUID file. Used for Homebrew user counting. Can be deleted and
-  # recreated with no adverse effect (beyond our user counts being inflated).
+  HOMEBREW_GIT_CONFIG_FILE="$HOMEBREW_REPOSITORY/.git/config"
+
+  # User UUID file. This file has been deprecated in favor of
+  # homebrew.analyticsuuid in $HOMEBREW_REPOSITORY/.git/config. If detected,
+  # the UUID will be written to the aforementioned git config file, and the
+  # original UUID file will be removed.
   HOMEBREW_ANALYTICS_USER_UUID_FILE="$HOME/.homebrew_analytics_user_uuid"
+  if [[ -f "$HOMEBREW_ANALYTICS_USER_UUID_FILE" ]]
+  then
+    HOMEBREW_ANALYTICS_USER_UUID="$(<"$HOMEBREW_ANALYTICS_USER_UUID_FILE")"
+    if [[ -n "$HOMEBREW_ANALYTICS_USER_UUID" ]]
+    then
+      git config --file="$HOMEBREW_GIT_CONFIG_FILE" --replace-all homebrew.analyticsuuid "$HOMEBREW_ANALYTICS_USER_UUID"
+    fi
+    rm -f "$HOMEBREW_ANALYTICS_USER_UUID_FILE"
+  fi
```

Comment:
Can we pull this entire block into a separate function (call it, e.g., `migrate-legacy-uuid-file`) that is invoked from here? This will make the `setup-analytics` function more digestible (as we link to this file and want people to actually look at it) and reduce distraction from what is actually going on in here.